core/src/qrsignatures.c

```diff
@@ -0,0 +1,43 @@
+#include <unistd.h>
+#include <stdbool.h>
+#include <string.h>
+#include <stdio.h>
+
+#include "ecdsa.h"
+#include "curves.h"
+#include "memzero.h"
+#include "nist256p1.h"
+#include "qrsignatures.h"
+#include "log.h"
+#include "config.h"
+
+static const uint8_t QR_PUBKEY[65] = _QR_PUBKEY;
+bool check_qrsignature(const uint8_t * const data, size_t data_len, const uint8_t * const signature){
+  if (data_len == 0){
+    ERROR("Input length is zero.");
+    return false;
+  }
+  if (data == NULL){
+    ERROR("Input data is null.");
+    return false;
+  }
+  if(signature == NULL){
+    ERROR("Signature is null");
+    return false;
+  }
+
+  int result_verify = ecdsa_verify(
+    (const ecdsa_curve *)&nist256p1,
+    HASHER_SHA2,
+    (const uint8_t *)QR_PUBKEY,
+    signature,
+    data,
+    (uint32_t)data_len);
```

Comment:
If `uint32_t` is expected by `ecdsa_verify`, it might actually be better to declare the parameter `data_len` as `uint32_t`. Note that `size_t` is not a fixed size type, and in theory, truncation could happen by casting it to `uint32_t`.

Reply:
I started with that :)
The serialized bytes have a proto limit of 4000 bytes for which any type/any platform is good. Hope that is good enough ?

Reply:
The truncation is only a theoretical concern, indeed. On the other hand, using `uint32_t` for `data_len` would make a code reviewer to think less harder here. :)

I don't have a strong opinion in this particular case.